dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1287,11 +1293,16 @@ private void registerZenodoDOIForWorkflow(String zenodoAccessToken, Workflow wor
             }
         }
 
-        provisionWorkflowVersionUploadFiles(zendoClient, returnDeposit,
+        File uploadFile = provisionWorkflowVersionUploadFiles(zendoClient, returnDeposit,
                 depositionID, workflow, workflowVersion);
 
         putDepositionOnZenodo(depositApi, depositMetadata, depositionID);
 
+        // Delete the zip file that contains the workflow files after it has been uploaded
+        // TODO: will the file be deleted by the deleteOnExit call in provisionWorkflowVersionUploadFiles
+        //  if putDepositionOnZenodo throws an exception?
+        uploadFile.delete();
```

Comment:
This should go in a finally in case anything throws an exception.

Reply:
done and refactored code